semsim/process_ontology.py

```diff
@@ -0,0 +1,64 @@
+"""Process ontology and retrieve pairwise similarities."""
+from collections import Counter
+
+import pandas as pd
+
+from .compute_pairwise_similarities import (compute_pairwise_ancestors_jaccard,
+                                            compute_pairwise_resnik)
+
+GRAPE_DATA_MOD = "grape.datasets.kgobo"
+ANNOTATION_SOURCES = {
+    "HP": "http://purl.obolibrary.org/obo/hp/hpoa/phenotype.hpoa"}
+
+
+def get_similarities(
+    ontology: str, resnik_path: str, ancestors_jaccard_path: str
+):
+    """Compute and store similarities to the provided paths.
+
+    :param ontology: str, name of ontology to retrieve and process.
+    :param resnik_path: str, where to store the resnik pairwise similarities.
+    :param ancestors_jaccard_path: str, where to store the Ancestors Jaccard
+        pairwise similarities.
+    """
+    ontology = ontology.upper()
+    onto_graph_class = import_grape_class(ontology)
+
+    onto_graph = (
+        onto_graph_class(directed=True)
+        .filter_from_names(
+            edge_type_names_to_keep=["biolink:subclass_of"],
+            node_prefixes_to_keep=[f"{ontology}:"],
+        )
+        .to_transposed()
+        .remove_disconnected_nodes()
+    )
+
+    counts = dict(
+        Counter(
+            pd.read_csv(
+                ANNOTATION_SOURCES[ontology],
+                sep="\t",
+                skiprows=4,
+            ).HPO_ID
```

Comment:
We will need to generalize `HPO_ID` for other ontologies. Is there a pattern for this?

Reply:
We may have to hold off on that until we have another annotation file to work with.
It's just going to be the header of the column containing CURIEs for whatever the target ontology is, but even this instance is strange since it's the usual "HP" vs "HPO" problem.

Reply:
Hmm ... .makes sense. There's no guarantee that the column name will end with a `_ID` either right?

Reply:
One solution here would be to make an argument for the column containing the counts (here HPO_ID)